src/__experimental__/components/decimal/decimal.component.js

```diff
@@ -336,7 +223,14 @@ Decimal.propTypes = {
   /**
    * The decimal precision of the value in the input
    */
-  precision: PropTypes.number,
+  precision: (props, propName) => {
+    if (props.precision < 0 || props.precision > Decimal.maxPrecision) {
```

Comment:
I think you need to check if it's a integer number as well here

```suggestion
    if (!Number.isInteger(precision) || props.precision < 0 || props.precision > Decimal.maxPrecision) {
```